js/classes/entities/midground/npc/village/mama_chad.js

```diff
@@ -0,0 +1,141 @@
+class MamaChad {
+    /**
+     * @param {Vector} pos the position at which he should spawn. 
+     * @param {Conversation} convo The conversation that will show if PapaChad is interacted with.
+     */
+    constructor(pos, convo = null) {
+        /** The position of the Papa Chad (in the game world). */
+        this.pos = pos;
+        /** The velocity at which PapaChad is moving. */
+        this.velocity = new Vector(0, 0);
+
+        /** An associative array of the animations for this Papa Chad. Arranged [facing][action]. */
+        this.animations = [];
+        this.loadAnimations();
+        /** What way is the Papa Chad looking? */
+        this.facing = "right";
+        /** What is the Papa Chad doing? */
+        this.action = "trapped";
+        /** Used to check for collisions with other applicable entities. */
+        this.boundingBox = new BoundingBox(this.pos, PapaChad.SCALED_SIZE);
+        /** Used to check how to deal with collisions with other applicable entities. */
+        this.lastBoundingBox = this.boundingBox;
+        /** The conversation which will be displayed upon interacting with Papa Chad. */
+        this.conversation = convo;
+    };
+
+    /** The size, in pixels of the sprite ON THE SPRITESHEET. */
+    static get SIZE() {
+        return new Vector(32, 64);
+    }
+
+    /** How much bigger should the sprite be drawn on the canvas than it is on the spritesheet? */
+    static get SCALE() {
+        return 2.8;
+    };
+
+    /** This will be the size of Papa Chad ON THE CANVAS. */
+    static get SCALED_SIZE() {
+        return PapaChad.SCALED_SIZE
+    }
+
+    /** The filepath to Papa Chad's spritesheet. */
+    static get SPRITESHEET() {
+        return "./sprites/blacksmith.png";
+    };
+
+    /** Change what Papa Chad is doing and where it is. */
+    update() {
+        if (this.action === 'trapped') return;
```

Comment:
I feel like depending on how you plan to handle this it may be better to have separate classes for normal MamaChad and trapped MamaChad, since it seems to me like they may not have much overlap functionally.

Reply:
I figure after you defeat the wizard, mama.action would switch to idle and she'd fall down, freed. So prob good to keep her in one place.

Reply:
That sounds good to me.

Reply:
> I figure after you defeat the wizard, mama.action would switch to idle and she'd fall down, freed. So prob good to keep her in one place.

Yeah! I like that